test/system/src/main/java/io/pravega/test/system/framework/services/docker/DockerBasedService.java

```
@@ -0,0 +1,191 @@
+            serviceId = serviceList.get(0).id();
+        } catch (DockerException e) {
+            log.error("Unable to get service id", e);
+        }
+        return serviceId;
+    }
+
+    @Override
+    public boolean isRunning() {
+        boolean value = false;
+        int taskRunningCount = 0;
+        try {
+            //list the service with filter 'serviceName'
+            Service.Criteria criteria1 = Service.Criteria.builder().serviceName(serviceName).build();
+            List<Service> serviceList1 = Exceptions.handleInterrupted(() -> dockerClient.listServices(criteria1));
+            log.info("service list size {}", serviceList1.size());
+            Task.Criteria taskCriteria1 = Task.Criteria.builder().serviceName(serviceName).build();
+            List<Task> taskList1 = Exceptions.handleInterrupted(() ->  dockerClient.listTasks(taskCriteria1));
+            log.info("task list size {}", taskList1.size());
+
+            if (!taskList1.isEmpty()) {
+                for (int j = 0; j < taskList1.size(); j++) {
+                    log.info("task {}", taskList1.get(j).name());
+                    String state = taskList1.get(j).status().state();
+                    log.info("task state {}", state);
+                    if (state.equals(TaskStatus.TASK_STATE_RUNNING)) {
+                        taskRunningCount++;
+                    }
+                }
+            }
+
+            if (!serviceList1.isEmpty()) {
+                long replicas = Exceptions.handleInterrupted(() -> dockerClient.inspectService(serviceList1.get(0).id()).spec().mode().replicated().replicas());
+                log.info("replicas {}", replicas);
+                log.info("task running count {}", taskRunningCount);
+                if (((long) taskRunningCount) == replicas) {
+                    return true;
+                }
+            }
+        } catch (DockerException e) {
+            log.error("Unable to list docker services", e);
```

We should not suppress exceptions.

---

done.